test/test/has-own-property.js

```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+Object.prototype[ 'Answer to the Ultimate Question of Life, the Universe, and Everything' ] = 42
+
+var tap = require('../..')
+
+function error() {
+  throw new Error('fake')
+}
+
+tap.test('obj.hasOwnProperty(key) checks', function (test) {
+  test.throws(error, 'this fails without internal checks')
+  test.end()
+})
```

Comment:
There's no need for an output check in test/test/.  A top-level test at `test/has-own-property.js` would be sufficient, and would only have to do this:

```javascript
Object.prototype.foo = 42
require('../').pass('modifying Object.prototype did not break things')
```